src/plugins/dashboard/public/application/actions/filters_notification_popover_contents.tsx

```diff
@@ -0,0 +1,103 @@
+import { FilterableEmbeddable, IEmbeddable } from '@kbn/embeddable-plugin/public';
+import { FilterItems } from '@kbn/unified-search-plugin/public';
+import { I18nProvider } from '@kbn/i18n-react';
+import { css } from '@emotion/react';
+import {
+  type AggregateQuery,
+  type Filter,
+  getAggregateQueryMode,
+  isOfQueryType,
+} from '@kbn/es-query';
+
+import { FiltersNotificationActionContext } from './filters_notification_action';
+import { dashboardFilterNotificationAction } from '../../dashboard_strings';
+import { DashboardContainer } from '../embeddable';
+
+export interface FiltersNotificationProps {
+  context: FiltersNotificationActionContext;
+}
+
+export function FiltersNotificationPopoverContents({ context }: FiltersNotificationProps) {
+  const { embeddable } = context;
+  const [isLoading, setIsLoading] = useState(true);
+  const [filters, setFilters] = useState<Filter[]>([]);
+  const [queryString, setQueryString] = useState<string>('');
+  const [queryLanguage, setQueryLanguage] = useState<'sql' | 'esql' | undefined>();
+
+  const dataViews = useMemo(
+    () => (embeddable.getRoot() as DashboardContainer)?.getAllDataViews(),
+    [embeddable]
+  );
+
+  useMount(() => {
+    Promise.all([
+      (embeddable as IEmbeddable & FilterableEmbeddable).getFilters(),
+      (embeddable as IEmbeddable & FilterableEmbeddable).getQuery(),
+    ]).then(([embeddableFilters, embeddableQuery]) => {
+      setFilters(embeddableFilters);
+      if (embeddableQuery) {
+        if (isOfQueryType(embeddableQuery)) {
+          setQueryString(embeddableQuery.query as string);
+        } else {
+          const language = getAggregateQueryMode(embeddableQuery);
+          setQueryLanguage(language);
+          setQueryString(embeddableQuery[language as keyof AggregateQuery]);
+        }
+      }
+      setIsLoading(false);
+    });
+  });
+
+  return (
+    <>
+      {isLoading ? (
+        <EuiLoadingContent lines={3} />
+      ) : (
+        <EuiForm
+          component="div"
+          css={css`
+            min-width: 300px;
+          `}
+        >
+          {queryString !== '' && (
+            <EuiFormRow
+              label={dashboardFilterNotificationAction.getQueryTitle()}
+              display="rowCompressed"
+            >
+              <EuiCodeBlock
+                language={queryLanguage}
+                paddingSize="none"
+                transparentBackground
+                aria-labelledby={`${dashboardFilterNotificationAction.getQueryTitle()}: ${queryString}`}
+                tabIndex={0} // focus so that keyboard controls will not skip over the code block
+              >
+                {queryString}
+              </EuiCodeBlock>
+            </EuiFormRow>
+          )}
+          {filters && filters.length > 0 && (
+            <EuiFormRow label={dashboardFilterNotificationAction.getFiltersTitle()}>
+              <EuiFlexGroup wrap={true} gutterSize="xs">
+                <I18nProvider>
```

Comment:
Why is I18nProvider nested here? Would it make sense to nest the entire FiltersNotificationPopover component in I18nProvider in FiltersNotificationAction so all i18n strings have a provider?

Reply:
Oh! Good point. The `FilterItems` component was getting angry that it wasn't wrapped in the `i18n` provider, so I just threw it there to fix the error. Totally right that it should go around the entire `FiltersNotificationAction` component - did this in bffd987c31a5b7c8d43de7126039a62db24a8c9e :+1:

Thanks for catching that!